regexp/regexp.go

```diff
@@ -0,0 +1,110 @@
+package regexp
+
+import (
+	"regexp"
+	"runtime"
+	"sync"
+	"weak"
+)
+
+// "Interns" compilation of Regular Expressions.  If two regexs with the same pattern are compiled, the result
+// is the same *regexp.Regexp.  This avoids the compilation cost but more importantly the memory usage.
+//
+// Regular expressions produced from this package are backed by a form of weak-valued map, upon a regexp becoming
+// unreachable, they will be eventually removed from the map and memory reclaimed.
+
+var (
+	weakMap      = make(map[string]weak.Pointer[regexp.Regexp])
+	posixWeakMap = make(map[string]weak.Pointer[regexp.Regexp])
+	reverseMap   = make(map[weak.Pointer[regexp.Regexp]]string)
+	l            sync.RWMutex
```

Comment:
can't this be a normal `Mutex`?

Reply:
Fixed in 071bc6fb408a627c06f061f6288efeafedae4b99